e2e_test.go

```diff
@@ -0,0 +1,1071 @@
+
+	// Exists
+
+	t.Run(`Test for Exists operation`, func(t *testing.T) {
+		beforeEach()
+		defer afterEach()
+
+		req := &payload.Object_ID{
+			Id: data[0].ID,
+		}
+
+		result, err := client.Exists(ctx, req)
+		if err != nil {
+			t.Errorf("error: %s", err)
+		}
+
+		if result.GetId() == "" {
+			t.Error("expected ID is not found.")
+		}
+	})
+
+	// GetObject
+	t.Run(`Test for GetObject operation`, func(t *testing.T) {
+		beforeEach()
+		defer afterEach()
+
+		req := &payload.Object_VectorRequest{
+			Id: &payload.Object_ID{
+				Id: data[0].ID,
+			},
+		}
+
+		result, err := client.GetObject(ctx, req)
+		if err != nil {
+			t.Errorf("error: %s", err)
+		}
+
+		if result.GetId() != data[0].ID {
+			t.Error("returned ID is not same as expected")
+		}
+
+		for i, elem := range result.GetVector() {
+			if elem != data[0].Vector[i] {
+				t.Error("returned vector is not same as expected")
+			}
+		}
+	})
+
+	t.Run(`Test for StreamGetObject operation`, func(t *testing.T) {
+		beforeEach()
+		defer afterEach()
+
+		stream, err := client.StreamGetObject(ctx)
+		if err != nil {
+			t.Fatalf("error: %s", err)
+		}
+
+		wg := sync.WaitGroup{}
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+
+			for {
+				result, err := stream.Recv()
+				if err != nil {
+					if err == io.EOF {
+						return
+					}
+
+					t.Errorf("error: %s", err)
+
+					continue
+				}
+
+				if result == nil {
+					t.Error("error: returned result is nil")
+					continue
+				}
+
+				res := result.GetVector()
+				if res == nil {
+					if result.GetStatus().GetCode() != 0 {
+						t.Errorf("result code is %d", result.GetStatus().GetCode())
+						continue
+					}
+
+					t.Error("location is nil")
+					continue
+				}
+
+				if res.GetId() == "" {
+					t.Error("returned ID is empty")
+				}
+
+				if len(res.GetVector()) == 0 {
+					t.Error("returned vector is empty")
+				}
+			}
+		}()
+
+		for i := 0; i < 11; i++ {
+			req := &payload.Object_VectorRequest{
+				Id: &payload.Object_ID{
+					Id: data[i].ID,
+				},
+			}
+
+			err = stream.Send(req)
+			if err != nil {
+				t.Errorf("error: %s", err)
+			}
+		}
+
+		stream.CloseSend()
+
+		wg.Wait()
+	})
+
+	// Search
+
+	t.Run(`Test for Search operation`, func(t *testing.T) {
+		beforeEach()
+		defer afterEach()
+
+		req := &payload.Search_Request{
+			Vector: data[0].Vector,
+			Config: &payload.Search_Config{
+				Num:     3,
+				Radius:  -1.0,
+				Epsilon: 0.1,
+				Timeout: 3000000000,
+			},
+		}
+
+		result, err := client.Search(ctx, req)
+		if err != nil {
+			t.Errorf("error: %s", err)
+		}
+
+		if result == nil {
+			t.Fatal("result is nil")
+		}
+
+		if len(result.GetResults()) != 3 {
+			t.Errorf("result length is not expected. received: %d, expected: %d", len(result.GetResults()), 3)
+		}
+	})
+
+	t.Run(`Test for MultiSearch operation`, func(t *testing.T) {
+		beforeEach()
+		defer afterEach()
+
+		cfg := &payload.Search_Config{
+			Num:     3,
+			Radius:  -1.0,
+			Epsilon: 0.1,
+			Timeout: 3000000000,
```

Comment:
Since this configuration is used from multiple lines as a same params, I think it would be better to use defaultSearchConfig or something.